src/zenml/model_registries/base_model_registry.py

```diff
@@ -0,0 +1,425 @@
+    ) -> Dict[str, Any]:
+        """Fills in ZenML metadata if not provided.
+
+        Args:
+            values: The values to validate.
+
+        Returns:
+            The validated values.
+        """
+        if values.get("zenml_metadata") is None:
+            values["zenml_metadata"] = ZenMLModelMetadata()
+        version_tags = values.get("version_tags")
+        if version_tags:
+            if "zenml_version" in version_tags.keys():
+                values["zenml_metadata"].zenml_version = version_tags[
+                    "zenml_version"
+                ]
+                values["version_tags"].pop("zenml_version")
+            if "zenml_pipeline_run_id" in version_tags.keys():
+                values["zenml_metadata"].zenml_pipeline_run_id = version_tags[
+                    "zenml_pipeline_run_id"
+                ]
+                values["version_tags"].pop("zenml_pipeline_run_id")
+            if "zenml_pipeline_name" in version_tags.keys():
+                values["zenml_metadata"].zenml_pipeline_name = version_tags[
+                    "zenml_pipeline_name"
+                ]
+                values["version_tags"].pop("zenml_pipeline_name")
+            if "zenml_step_name" in version_tags.keys():
+                values["zenml_metadata"].zenml_step_name = version_tags[
+                    "zenml_step_name"
+                ]
+                values["version_tags"].pop("zenml_step_name")
+        return values
+
+
+class BaseModelRegistryConfig(StackComponentConfig):
+    """Base config for model registries."""
+
+
+class BaseModelRegistry(StackComponent, ABC):
+    """Base class for all ZenML model registries."""
+
+    @property
+    def config(self) -> BaseModelRegistryConfig:
+        """Returns the config of the model registries.
+
+        Returns:
+            The config of the model registries.
+        """
+        return cast(BaseModelRegistryConfig, self._config)
+
+    # ---------
+    # Model Registration Methods
+    # ---------
+
+    @abstractmethod
+    def register_model(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> ModelRegistration:
+        """Registers a model in the model registry.
+
+        Args:
+            name: The name of the registered model.
+            description: The description of the registered model.
+            tags: The tags associated with the registered model.
+        """
+
+    @abstractmethod
+    def delete_model(
+        self,
+        name: str,
+    ) -> None:
+        """Deletes a registered model from the model registry.
+
+        Args:
+            name: The name of the registered model.
+        """
+
+    @abstractmethod
+    def update_model(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> ModelRegistration:
+        """Updates a registered model in the model registry.
+
```

Comment:
You should set out some rules and expectations in the abstraction concerning what happens if you don't have a model with that name in the registry. Do you expect a `zenml.exceptions.EntityExistsError` to be raised ?